src/terms/kinetic.jl

```diff
@@ -1,22 +1,66 @@
 """
-Kinetic energy: 1/2 sum_n f_n ∫ |∇ψn|^2.
+Kinetic energy: 1/2 sum_n f_n ∫ |∇ψn|^2 (standard)
+             or 1/2 sum_n f_n ∫ blowup(-i∇Ψ)^2 (modified kinetic term).
 """
-struct Kinetic
-    scaling_factor::Real
+Base.@kwdef struct Kinetic{F}
+    scaling_factor::Real = 1
+    blowup::F = BlowupKineticEnergy() # blow-up function to smooth energy bands. Defaut is x↦x^2.
 end
-Kinetic(; scaling_factor=1) = Kinetic(scaling_factor)
-(kin::Kinetic)(basis) = TermKinetic(basis, kin.scaling_factor)
+
+(kin::Kinetic)(basis) = TermKinetic(basis, kin.scaling_factor, kin.blowup)
 function Base.show(io::IO, kin::Kinetic)
     fac = isone(kin.scaling_factor) ? "" : ", scaling_factor=$scaling_factor"
     print(io, "Kinetic($fac)")
 end
 
+"""
+Define different energy cutoff strategies.
+ADD REF when the paper is published.
+"""
+struct BlowupKineticEnergy end
+struct BlowupCHV end
+
+"""
+Default blow-up, x->x^2, corresponding to the standard kinetic energies.
+"""
+(blowup::BlowupKineticEnergy)(x) = sum(abs2, x)
+
+"""
+Blow-up function as proposed in [REF paper Cancès, Hassan, Vidal to be submitted]
+Parameters have been computed to ensure C^2 regularity of the energies bands
+away from crossings and Lipschitz continuity at crossings.
+TODO: find a better C^2 interpolation.
+"""
+function (blowup::BlowupCHV)(x)
+    @assert (0 ≤ x < 1) "The blow-up function is defined on [0,1)"
+    # Define the three parts of the blow-up function
+    g1(x) = sum(abs2, x)
+    # C^2 interpolation part. The coefficients have been obtained by solving a
+    # linear system.
+    C = [7449.468883604184, -50230.90030429237, 135207.0860245428,
+         -181568.08750464107, 121624.06499834878, -32503.74733526006]
+    g2(x) = C'*((x * ones(6)) .^(0:5))
+    # Blow-up part
+    Ca(a) = (3/2)*(a^2)*(1-a)^(2)
+    a_opti = 0.1792973510040141
+    g3(x) = Ca(a_opti)/( (1-x)^(2) )
+
+    # Assemble all parts
+    x1, x2 = 0.7, 0.8
+    (0 ≤ x < x1)   && return g1(x)
+    (x1 ≤ x < x2)  && return g2(x)
+    (x2 ≤ x < 1)   && return g3(x)
+    (x==1) && return 1e6 # Handle |G+k|^2 = E_cut case
+end
+
 struct TermKinetic <: Term
     scaling_factor::Real  # scaling factor, absorbed into kinetic_energies
-    kinetic_energies::Vector{<:AbstractVector}  # kinetic energy 1/2|G+k|^2 for every kpoint
+    # kinetic energies 1/2(k+G)^2 (or Ecut*blowup(|k+G|/√(2Ecut)) for energy
+    # cutoff smearing methods) for each k-point.
+    kinetic_energies::Vector{<:AbstractVector}
 end
-function TermKinetic(basis::PlaneWaveBasis{T}, scaling_factor) where {T}
-    kinetic_energies = [[T(scaling_factor) * sum(abs2, Gk) / 2
+function TermKinetic(basis::PlaneWaveBasis{T}, scaling_factor, blowup) where {T}
+    kinetic_energies = [[T(scaling_factor) * basis.Ecut * blowup(norm(Gk)/√(2Ecut))*
```

Comment:
why is there a basis.Ecut * here, and why is there a * at the end (this is probably the source of the CI issues)